provider/gcp/gcp_instance.go

```diff
@@ -117,6 +118,61 @@ func (p *GCloud) CreateInstance(ctx *lepton.Context) error {
 		}
 	}
 
+	// Attach disk(s) upon instance creation if RunConfig.AttachVolumeOnInstanceCreate is set
+	for mntSrc, mntDst := range c.Mounts {
+		if !c.RunConfig.AttachVolumeOnInstanceCreate {
+			continue
+		}
```

Comment:
This check can be done just once, if the attach option is false we don't need to execute the for loop.